tool/release.rb

```diff
@@ -122,7 +127,13 @@ def self.install_dependencies!
   end
 
   def self.for_bundler(version)
-    rubygems_version = Gem::Version.new(version).segments.map.with_index {|s, i| i == 0 ? s + 1 : s }.join(".")
+    segments = Gem::Version.new(version).segments
+    rubygems_version = if segments[0] >= 4
+      segments.join(".")
+    else
+      segments.map.with_index {|s, i| i == 0 ? s + 1 : s }.join(".")
+    end
+    rubygems_version.gsub(/([a-z])\.(\d)/i, '\1\2')
```

Comment:
This is needless because we don't use `gsub!` nor assign result here:

```suggestion
```

(And `Gem::Version.new` produces the same results for both of `4.0.0.beta1` and `4.0.0.beta.1`.)